lib/ansible/playbook/base.py

```diff
@@ -631,7 +631,7 @@ def post_validate(self, templar):
                     continue
                 else:
                     raise AnsibleParserError("the field '%s' is required but was not set" % name)
-            elif not attribute.always_post_validate and self.__class__.__name__ not in ('Task', 'Handler', 'PlayContext'):
+            elif not attribute.always_post_validate and self.__class__.__name__ not in ('Task', 'Handler', 'PlayContext', 'LoopControl'):
```

Comment:
I'm not sure this should be place to set the classes, since we allow for `isa=class` and `class_type=` in the FA definition it seems we should have a system that depends on allowable values there

note: not putting the onus on this PR, just looking into making the system more flexible for other keywords to use a playbook object and still be able to handle normalish inheritance and validation